src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1482,6 +1482,13 @@ private string GetLocalDebugName(ILocalSymbolInternal local, out LocalDebugId lo
                 return null;
             }
 
+            // Skip naming locals generated by instrumentation as they might not have a local syntax offset.
+            // Locals generated by instrumentation might exist in methods which do not contain a body (auto property initializers).
+            if (local.SynthesizedKind == SynthesizedLocalKind.InstrumentationPayload)
+            {
+                return null;
```

Comment:
If we don't do this then a NRE is thrown, when calling the `_method.GetLocalDebugName` (as  autoproperty `_method` does not contain a BodySyntax).
This is the exception's stack trace:

```
CodeGen\EmitStatement.cs(1490,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.GetLocalDebugName(ILocalSymbolInternal local, LocalDebugId& localId)
CodeGen\EmitStatement.cs(1440,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.DefineLocal(LocalSymbol local, CSharpSyntaxNode syntaxNode)
CodeGen\EmitStatement.cs(603,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.EmitBlock(BoundBlock block)
CodeGen\EmitStatement.cs(23,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.EmitStatement(BoundStatement statement)
CodeGen\EmitStatement.cs(102,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.EmitStatementList(BoundStatementList list)
CodeGen\EmitStatement.cs(39,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.EmitStatement(BoundStatement statement)
CodeGen\CodeGenerator.cs(255,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.GenerateImpl()
CodeGen\CodeGenerator.cs(190,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.Generate()
Compiler\MethodCompiler.cs(1340,0): at Microsoft.CodeAnalysis.CSharp.MethodCompiler.GenerateMethodBody(PEModuleBuilder moduleBuilder, MethodSymbol method, Int32 methodOrdinal, BoundStatement block, ImmutableArray`1 lambdaDebugInfo, ImmutableArray`1 closureDebugInfo, StateMachineTypeSymbol stateMachineTypeOpt, VariableSlotAllocator variableSlotAllocatorOpt, DiagnosticBag diagnostics, DebugDocumentProvider debugDocumentProvider, ImportChain importChainOpt, Boolean emittingPdb, ImmutableArray`1 dynamicAnalysisSpans)
Compiler\MethodCompiler.cs(1110,0): at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethod(MethodSymbol methodSymbol, Int32 methodOrdinal, ProcessedFieldInitializers& processedInitializers, SynthesizedSubmissionFields previousSubmissionFields, TypeCompilationState compilationState)
Compiler\MethodCompiler.cs(433,0): at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(NamedTypeSymbol containingType)
Compiler\MethodCompiler.cs(323,0): at Microsoft.CodeAnalysis.CSharp.MethodCompiler.<>c__DisplayClass20_0.<CompileNamedTypeAsTask>b__0()
InternalUtilities\UICultureUtilities.cs(136,0): at Roslyn.Utilities.UICultureUtilities.<>c__DisplayClass5_0.<WithCurrentUICulture>b__0()
```